src/Microsoft.ML/CSharpApi.cs

```diff
@@ -15626,36 +15626,37 @@ public sealed class UniformRandomAutoMlEngine : AutoMlEngine
 
         public abstract class AutoMlStateBase : ComponentKind {}
 
-        public enum AutoInferenceAutoMlMlStateArgumentsMetrics
+        public enum PipelineSweeperSupportedMetricsMetrics
         {
             Auc = 0,
             AccuracyMicro = 1,
             AccuracyMacro = 2,
-            L2 = 3,
-            F1 = 4,
-            AuPrc = 5,
-            TopKAccuracy = 6,
-            Rms = 7,
-            LossFn = 8,
-            RSquared = 9,
-            LogLoss = 10,
-            LogLossReduction = 11,
-            Ndcg = 12,
-            Dcg = 13,
-            PositivePrecision = 14,
-            PositiveRecall = 15,
-            NegativePrecision = 16,
-            NegativeRecall = 17,
-            DrAtK = 18,
-            DrAtPFpr = 19,
-            DrAtNumPos = 20,
-            NumAnomalies = 21,
-            ThreshAtK = 22,
-            ThreshAtP = 23,
-            ThreshAtNumPos = 24,
-            Nmi = 25,
-            AvgMinScore = 26,
-            Dbi = 27
+            L1 = 3,
+            L2 = 4,
+            F1 = 5,
+            AuPrc = 6,
+            TopKAccuracy = 7,
+            Rms = 8,
+            LossFn = 9,
+            RSquared = 10,
+            LogLoss = 11,
+            LogLossReduction = 12,
+            Ndcg = 13,
+            Dcg = 14,
+            PositivePrecision = 15,
+            PositiveRecall = 16,
+            NegativePrecision = 17,
+            NegativeRecall = 18,
+            DrAtK = 19,
+            DrAtPFpr = 20,
+            DrAtNumPos = 21,
+            NumAnomalies = 22,
+            ThreshAtK = 23,
+            ThreshAtP = 24,
+            ThreshAtNumPos = 25,
+            Nmi = 26,
+            AvgMinScore = 27,
+            Dbi = 28
```

Comment:
Looks like they are auto generated by the CSharp API generator

---
In reply to: [203217022](https://github.com/dotnet/machinelearning/pull/539#discussion_r203217022) [](ancestors = 203217022)

Reply:
If we serialize the enum, I think we'd want a fixed number. This could be serialized as a state object for the sweep so that the user can resume the sweep w/ the history of the sweep stored on disk.

Reply:
Not sure if this is a general comment , or something that needs to be addressed ?

Btw, this is is auto generated file . The Enum you see here is defined in PipelineSweeperSupportedMetrics.cs

---
In reply to: [203267499](https://github.com/dotnet/machinelearning/pull/539#discussion_r203267499) [](ancestors = 203267499)

Reply:
It's a topic to consider for the future. 
When we serialize the enum, we'll either have to serialize the numeric form or the text form. A numeric form will be more stable, and for that we'd have to assign the numbers in the primary definition of the enum.